src/status_im/contexts/wallet/wallet_connect/rpc.cljs

```diff
@@ -0,0 +1,61 @@
+(ns status-im.contexts.wallet.wallet-connect.rpc
+  (:require [oops.core :as oops]
+            [promesa.core :as promesa]
+            [status-im.common.json-rpc.events :as rpc-events]
+            [status-im.constants :as constants]
+            [utils.transforms :as transforms]))
+
+(defn- call-rpc
+  "Helper to handle RPC calls to status-go as promises"
+  [method & args]
+  (promesa/create
+   (fn [p-resolve p-reject]
+     (rpc-events/call {:method      method
+                       :params      (vec args)
+                       :on-success  p-resolve
+                       :on-error    p-reject
+                       :js-response true}))))
+
+(defn wallet-build-transaction
```

Comment:
> By the way, I remember the argument to use the long alias promesa was because people weren't accustomed to it. One of these days I think we should rename it back to p because that's what everybody using promesa does.

Yep, I think we should. At least I hope it's familiar enough to use the shorter version. @J-Son89 WDYT?